common.mk

```diff
@@ -139,15 +152,29 @@ else
 
 endif
 
+# compiler=clang Enables compilation with clang. 
+
+ifeq ($(compiler), clang)
+	BIN_SUFFIX := sm$(SM_ARCH)_clang_$(NVCC_VERSION)
```

Comment:
It makes sense to annotate which version of the CUDA runtime libraries you're using, though this code is a little misleading as it's using a variable called NVCC_VERSION (as opposed to something like CUDA_VERSION), and also because it could mislead someone into thinking that they're somehow using clang version 9.0 (or whatever CUDA version is being used), even though they're of course not.

You could consider doing something like `clang_$(CLANG_VERSION)_cuda_$(NVCC_VERSION)` instead.

Reply:
I'll buy the argument that build should be annotated with CUDA version as that's really what we care about here. Specific version of nvcc or clang is largely irrelevant here. I'll update the patch to reflect CUDA version.